src/services/GraphsDFG.jl

```diff
@@ -0,0 +1,432 @@
+module GraphsJl
+
+using Graphs
+using Requires
+using DocStringExtensions
+using DistributedFactorGraphs
+
+"""
+$(SIGNATURES)
+Encapsulation structure for a DFGNode (Variable or Factor) in Graphs.jl graph.
+"""
+mutable struct GraphsNode
+    index::Int
+    dfgNode::DFGNode
+end
+const FGType = Graphs.GenericIncidenceList{GraphsNode,Graphs.Edge{GraphsNode},Dict{Int,GraphsNode},Dict{Int,Array{Graphs.Edge{GraphsNode},1}}}
+
+export GraphsDFG
+export addVariable!
+export addFactor!
+export ls, lsf, getVariables, getFactors
+export getVariable, getFactor
+export updateVariable!, updateFactor!
+export deleteVariable!, deleteFactor!
+export getAdjacencyMatrix
+export getAdjacencyMatrixDataFrame
+export getNeighbors
+export getSubgraphAroundNode
+export getSubgraph
+export isFullyConnected
+
+mutable struct GraphsDFG <: AbstractDFG
+    g::FGType
+    description::String
+    nodeCounter::Int64
+    labelDict::Dict{Symbol, Int64}
+end
+
+"""
+    $(SIGNATURES)
+Create a new in-memory Graphs.jl-based DFG factor graph.
+"""
+GraphsDFG() = GraphsDFG(Graphs.incdict(GraphsNode,is_directed=false), "Graphs.jl implementation", 0, Dict{Symbol, Int64}())
+
+"""
+    $(SIGNATURES)
+Add a DFGVariable to a DFG.
+"""
+function addVariable!(dfg::GraphsDFG, variable::DFGVariable)::Bool
+    if haskey(dfg.labelDict, variable.label)
+        error("Variable '$(variable.label)' already exists in the factor graph")
+    end
+    dfg.nodeCounter += 1
+    variable._internalId = dfg.nodeCounter
+    v = GraphsNode(dfg.nodeCounter, variable)
+    Graphs.add_vertex!(dfg.g, v)
+    push!(dfg.labelDict, variable.label=>variable._internalId)
+    return true
+end
+
+"""
+    $(SIGNATURES)
+Add a DFGFactor to a DFG.
+"""
+function addFactor!(dfg::GraphsDFG, factor::DFGFactor, variables::Vector{DFGVariable})::Bool
+    if haskey(dfg.labelDict, factor.label)
+        error("Factor '$(factor.label)' already exists in the factor graph")
+    end
+    for v in variables
+        if !(v.label in keys(dfg.labelDict))
+            error("Variable '$(v.label)' not found in graph when creating Factor '$(factor.label)'")
+        end
+    end
+    dfg.nodeCounter += 1
+    factor._internalId = dfg.nodeCounter
+    fNode = GraphsNode(dfg.nodeCounter, factor)
+    f = Graphs.add_vertex!(dfg.g, fNode)
+    # Add index
+    push!(dfg.labelDict, factor.label=>factor._internalId)
+    # Add the edges...
+    for variable in variables
+        v = dfg.g.vertices[variable._internalId]
+        edge = Graphs.make_edge(dfg.g, f, v)
+        Graphs.add_edge!(dfg.g, edge)
+    end
+    return true
+end
+
+"""
+    $(SIGNATURES)
+Get a DFGVariable from a DFG using its underlying integer ID.
+"""
+function getVariable(dfg::GraphsDFG, variableId::Int64)::DFGVariable
+    @warn "This may be slow, rather use by getVariable(dfg, label)"
```

Comment:
Is this from test results?  My very brief experience was that Symbol lookups on a dict where much much slower than Int key lookups, and strong lookups were much slower than Symbol.  We should confirm.

Reply:
The worry isn't actually on the lookup itself, rather on the check to see if the variable is there. Probably O(n) to check if it's in the values of the dictionary. If this is used, then it's probably a good idea to have a idDict
```
    if !(variableId in values(dfg.labelDict))
        error("Variable ID '$(variableId)' does not exist in the factor graph")
    end
```